src/main/java/parkinglot/ParkingLotTest.java

```diff
@@ -22,45 +30,118 @@ public static void main(String[] args) throws InterruptedException {
+//        }
+//        TimeUnit.SECONDS.sleep(timeout);
+//
+//        // park vehicle
+//        try {
+//            System.out.println(parkingLot.parkVehicleOnParkingLot(car2));
+//        } catch (ParkingLotIsFullException e) {
+//            System.out.println(e.getMessage());
+//        }
+//        TimeUnit.SECONDS.sleep(timeout);
+
+        String parkingTicketNumber = null;
 
         // unpark vehicle
-        String parkingTicketNumber = parkingLot.getParkingTicketNumber(car1);
-        System.out.println(parkingLot.unparkVehicleFromParkingLot(car1, parkingTicketNumber));
+        try {
+            parkingTicketNumber = parkingLot.getParkingTicketNumber(car1);
+            try {
+                System.out.println(parkingLot.unparkVehicleFromParkingLot(car1, parkingTicketNumber));
+            } catch (VehicleNotFoundOnParkingLotException | ParkingLotIsEmptyException e) {
+                System.out.println(e.getMessage());
+            }
+        } catch (VehicleNotFoundOnParkingLotException | ParkingTicketNumberNotFoundException e) {
+            System.out.println(e.getMessage());
+        }
 
-
-        System.out.println(parkingLot.parkVehicleOnParkingLot(car3));
+        // unpark vehicle
+        try {
+            parkingTicketNumber = parkingLot.getParkingTicketNumber(car1);
+            try {
+                System.out.println(parkingLot.unparkVehicleFromParkingLot(car1, parkingTicketNumber));
+            } catch (VehicleNotFoundOnParkingLotException | ParkingLotIsEmptyException e) {
+                System.out.println(e.getMessage());
+            }
+        } catch (VehicleNotFoundOnParkingLotException | ParkingTicketNumberNotFoundException e) {
+            System.out.println(e.getMessage());
+        }
+
+        // park vehicle
+        try {
+            System.out.println(parkingLot.parkVehicleOnParkingLot(car3));
+        } catch (ParkingLotIsFullException e) {
+            System.out.println(e.getMessage());
+        }
         TimeUnit.SECONDS.sleep(timeout);
 
-        System.out.println(parkingLot.parkVehicleOnParkingLot(motorcycle1));
+        // park vehicle
+        try {
+            System.out.println(parkingLot.parkVehicleOnParkingLot(motorcycle1));
+        } catch (ParkingLotIsFullException e) {
+            System.out.println(e.getMessage());
+        }
         TimeUnit.SECONDS.sleep(timeout);
 
-        System.out.println(parkingLot.parkVehicleOnParkingLot(bus3));
+
+        // park vehicle
+        try {
+            System.out.println(parkingLot.parkVehicleOnParkingLot(bus3));
+        } catch (ParkingLotIsFullException e) {
+            System.out.println(e.getMessage());
+        }
         TimeUnit.SECONDS.sleep(timeout);
+
+
         // unpark vehicle
-        parkingTicketNumber = parkingLot.getParkingTicketNumber(motorcycle1);
-        System.out.println(parkingLot.unparkVehicleFromParkingLot(motorcycle1, parkingTicketNumber));
+        try {
```

Comment:
maybe it is better to encapsulate those try-catch blocks within the methods, not to show it in ParkingLotTest

Reply:
fixed